cmd/makedocs/main.go

```diff
@@ -12,38 +17,60 @@ import (
 
 const (
 	docsPath      = "../../docs/"
-	checkersPath  = docsPath + "checkers/"
 	templatesPath = docsPath + "templates/"
+	checkersPath  = "../../lint/"
 )
 
 type checker struct {
 	Name             string
 	ShortDescription string
 	Description      string
+	Before           string
+	After            string
+	Note             string
 	SyntaxOnly       bool
 	Experimental     bool
 	VeryOpinionated  bool
 }
 
+var checkers []checker
+
 func main() {
 	tmpl, err := template.ParseFiles(templatesPath + "overview.md.tmpl")
 	if err != nil {
 		log.Fatal(err)
 	}
-	checkers := []checker{}
+	pkgs, err := parser.ParseDir(&token.FileSet{}, checkersPath,
+		func(inf os.FileInfo) bool {
+			return strings.HasSuffix(inf.Name(), "_checker.go")
+		}, parser.ParseComments)
+	if err != nil {
+		log.Fatal(err)
+	}
+
 	for _, r := range lint.RuleList() {
-		desc, err := getDesc(r.Name())
-		if err != nil {
-			log.Fatal(r.Name())
+		log.Printf("parsing %s\n", r.Name())
+
+		f, ok := pkgs["lint"].Files[fmt.Sprintf("%s%s_checker.go", checkersPath, r.Name())]
+		if !ok {
+
+			log.Printf("File not found: %s%s_checker.go", checkersPath, r.Name())
+			continue
+		}
+		c := checker{
+			Name:            r.Name(),
+			SyntaxOnly:      r.SyntaxOnly,
+			Experimental:    r.Experimental,
+			VeryOpinionated: r.VeryOpinionated,
```

Comment:
You could store `*lint.Rule` inside `checker` and do just: `c := checker{rule: rule}`.
Current approach also makes sense, so this is not a request for changes.